libvast/src/system/index.cpp

```diff
@@ -994,8 +977,9 @@ index(index_actor::stateful_pointer<index_state> self,
             auto delta = std::chrono::steady_clock::now() - start;
             VAST_TRACEPOINT(query_meta_index, query_id.as_u64().first, total,
                             delta.count());
-            respond(query_id, detail::narrow<uint32_t>(total), scheduled);
-            rp.delegate(static_cast<index_actor>(self), query_id, scheduled);
+            rp.deliver(query_id, detail::narrow<uint32_t>(total), scheduled);
+            caf::send_as(client, static_cast<index_actor>(self), query_id,
+                         scheduled);
```

Comment:
Not sure I like this better than the old approach: This is a hand-rolled delegate, except that it does not keep the message ID and the priority.

Can't you create a second response promise before this request-then you're in which you return from the handler, and delegate that here?

Reply:
That doesn't work. I already have to return the other `rp` from the outer handler. I tried without returning the extra promise but it the response doesn't arrive at the exporter.

Reply:
You're right.